text/0000-rustfmt-stability.md

```diff
@@ -0,0 +1,218 @@
+
+#### API clients
+
+Can control versioning using Cargo, but there is likely to be pressure from end-users to have an up to date version. API breaking changes would cause build errors. Formatting changes might break tests.
+
+#### Options
+
+Rustfmt can be configured with [many options](https://github.com/rust-lang-nursery/rustfmt/blob/master/Configurations.md) from the command line and a config file (rustfmt.toml). Options can be stable or unstable. Unstable options can only be used on the nightly toolchain and require opt-in with the `--unstable-features` command line flag. All options have default values and users are strongly encouraged to use these defaults.
+
+There is currently an unstable `required_version` option which enforces that the program is being formatted with a given version of Rustfmt, however, there is no mechanism to get the specified version.
+
+
+### Distribution and versioning
+
+Rustfmt can be built and run [from source](https://github.com/rust-lang-nursery/rustfmt), but requires a nightly toolchain. It can be used as a library or installed via Cargo. It is versioned in the usual way (currently 0.6.0). There are two crates on crates.io - `rustfmt-nightly` is up to date and requires a nightly toolchain, `rustfmt` is deprecated.
+
+If using Rustfmt as a tool, the recommended way to install is via Rustup (`rustup component add rustfmt-preview`). This method does not require a nightly toolchain. Versioning is linked to the Rust toolchain (Rustfmt is not meaningfully versioned outside of the Rust version). The version of Rustfmt available on the nightly channel depends on the version of the rustfmt submodule in the Rust repo. This is manually updated approximately once per week. The version of Rustfmt available on beta and stable is the version on nightly when it became beta. Updates to rustfmt on the beta channel happen occasionally.
+
+If using Rustup, then there is no way to get a specific version of Rustfmt, only the version associated with a specific version of Rust.
+
+A common way to use Rustfmt is in an editor via the RLS. The RLS is primarily distributed via Rustup. When installed in this way, the version of Rustfmt used is the same as if Rustfmt were installed directly via Rustup (note that this is the version in the Rust repo submodule and not necessarily the same version as indicated by the RLS's Cargo.toml).
+
+
+## Definition of changes
+
+In this section we define what constitutes different kinds of breaking change for Rustfmt.
+
+
+### API breaking change
+
+A change that could cause a dependent crate not to build, could break a script using the executable, or breaks specification-level formatting compatibility. A formatting change in this category would be frustrating even for occasional users.
+
+Examples:
+
+* remove a stable option (config or command line)
+* remove or change some variants of a stable option
```

Comment:
Might be worth clarifying, this is strictly talking about the _existence_ of options and not their behavior.

As it stands this line is a bit confusing, at first I interpreted it as "changing how stable options format code", which is _not_ a breaking change as long as the options that are changed are not the _default_ ones.